### DISCUSSION TITLE
need to disable the  open file button

### DISCUSSION THREAD

**Discussion by emphyrio61** [sumatrapdfreader/sumatrapdf#2955] [Q&A]
I start Sumatra from a shell command in VBA with a pdf file location in the command text. But I cannot allow my users access to the file system. So I need to disable the file open button. If I start Sumatra in -restrict mode, the file open button is disabled.  However, that also disables  the copy and print functionality...  And that is too much. I just want to restrict the file open button and command. How can I do this?

Hope someone can help me with this.
BR, Emphyrio61

**Comment by emphyrio61** [comment]
I found a partial solution, by using the sumatrapdfrestrict.ini file and disable DiskAccess. That works, but the downside is that it also disables the print dialog. Hence the users can no longer select the printer. The file is sent to the default printer directly. I can live with that, but it would be nice if you had the option to disable disk access, but still allow the print dialog to appear... Is there a work around?

**Comment by GitHubRulesOK** [comment]
read the manual for what actions can be disabled or allowed but some may be mutual thus one may not contradict the other so try different ones, for copy and paste you can enable just that one

note that current version 3.3.+ has abilities that may via contortions bypass restrictions, (generally most abilities are restrictions aware, but calling browser could be abused and annotation saving naturally allows external access) thus an older version 3.2 may be preferred.

if I set `DiskAccess = 0` then I can not open files but  to print to my virtual printer that will itself allow disk access to save the output, it is part of windows natural behavior, thus you need to remove from system any virtual file printer such as print to pdf 

<strike>The print current page function can be a separate command file invocation to keep it from the internal access requirement. You would need to build a command that itself is restricted in print ability and assign to `Open in Restricted printing.` but I have not tested if that is possible!</strike> no that's not possible as open in is naturally blocked.

**Comment by emphyrio61** [comment]
I see what you mean. Thank  you for this information. I will revert to version 3.2 then. I wonder why the developers keep putting in these options that rarely anyone will use and renders the product unsuitable for professional environments. And if you do put in these options, please make  it possible to switch them off...!

**Reply by emphyrio61** [reply]
I installed version 3.2. The annotation feature is gone, so that is good. But now the buttons to rotate the document are gone :( I can still rotate the document with the [ ] keys, but I would - for my users - prefer to show this option on the UI. Is there a way to show rotation buttons in version 3.2?

**Reply by GitHubRulesOK** [reply]
NO the additional buttons were added with the change of icons from bmp to svg you can only change the 3.2 bmp appearance or size by substitution. some users forked to swap pairs with other button actions and icons but that's into compilation territory, defeating permissive commercial use. It is by the virtue of using public available binaries that permits corporate personal usage. It is not a fully commercially free oss developer product. however you could use 3rd party AutoHotKey assignments with replacement image to achieve that goal and still be within permitted use :-) the side benefit is the possibility of collecting filename/page number for custom printing, but that's a tricky fiddle using code by Nod5.

**Reply by emphyrio61** [reply]
Ok, I understand. It is not a big deal. But pity that the annotations feature circumvents the DiskAccess = 0 setting. It should not be that way. It would be nice if the annotations feature would be switched OFF when DiskAccess = 0. That would immediately take care of the annotations issue.

**Reply by GitHubRulesOK** [reply]
I am not developer so 
@kjk this point of annotations need disk access has also been reflected in a different request to disable accidental invocation of dragging
